lib/perfdata/influxdbchecktask.cpp

```diff
@@ -0,0 +1,141 @@
+)
+{
+	if (Checkable::ExecuteCommandProcessFinishedHandler) {
+		double now = Utility::GetTime();
+		ProcessResult pr;
+		pr.PID = -1;
+		pr.Output = std::move(output);
+		pr.ExecutionStart = now;
+		pr.ExecutionEnd = now;
+		pr.ExitStatus = state;
+
+		Checkable::ExecuteCommandProcessFinishedHandler(commandObj->GetName(), pr);
+	} else {
+		cr->SetState(state);
+		cr->SetOutput(output);
+		checkable->ProcessCheckResult(cr);
+	}
+}
+
+void InfluxdbCheckTask::ScriptFunc(
+	const Checkable::Ptr& checkable, const CheckResult::Ptr& cr,
+	const Dictionary::Ptr& resolvedMacros, bool useResolvedMacros
+)
+{
+	ServiceState state = ServiceOK;
+	CheckCommand::Ptr commandObj = CheckCommand::ExecuteOverride ? CheckCommand::ExecuteOverride : checkable->GetCheckCommand();
+
+	Host::Ptr host;
+	Service::Ptr service;
+	tie(host, service) = GetHostService(checkable);
+
+	MacroProcessor::ResolverList resolvers;
+
+	if (MacroResolver::OverrideMacros) {
+		resolvers.emplace_back("override", MacroResolver::OverrideMacros);
+	}
+
+	if (service)
+		resolvers.emplace_back("service", service);
+	resolvers.emplace_back("host", host);
+	resolvers.emplace_back("command", commandObj);
+	resolvers.emplace_back("icinga", IcingaApplication::GetInstance());
+
+	String influxdbName = MacroProcessor::ResolveMacros("$influxdb_name$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	String missingQueriesWarning;
+	String missingQueriesCritical;
+
+	double queriesWarning = MacroProcessor::ResolveMacros("$influxdb_queries_warning$", resolvers, checkable->GetLastCheckResult(),
+		&missingQueriesWarning, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	double queriesCritical = MacroProcessor::ResolveMacros("$influxdb_queries_critical$", resolvers, checkable->GetLastCheckResult(),
+		&missingQueriesCritical, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	if (resolvedMacros && !useResolvedMacros)
+		return;
+
+	if (influxdbName.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "Attribute 'influxdb_name' must be set.");
+		return;
+	}
+
+	const InfluxdbWriter::Ptr& influxdbWriter = InfluxdbWriter::GetByName(influxdbName);
+
+	if (!influxdbWriter) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter '" + influxdbName + "' object does not exist.");
+		return;
+	}
+
+	if (influxdbWriter->IsPaused()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter is temporarily disabled on this cluster instance.", state);
+		return;
+	}
+
+	InfluxdbWriter::OnLastErrorMessageChanged.connect([influxdbWriter, checkable, commandObj, cr](
+		const InfluxdbWriter::Ptr&, const Value&) {
+		auto errorMessage = influxdbWriter->GetLastErrorMessage();
+
+		if (!errorMessage.IsEmpty()) {
+			ReportInfluxdbCheck(checkable, commandObj, cr, errorMessage, ServiceCritical);
+			m_ErrorOccurred = true;
+		}
+	});
```

Comment:
When do you think this callback fires?

Reply:
Well, whenever the ``last_error_message`` is changed.

Reply:
Does a user who set a check_interval of 5m wants a check result _whenever the last_error_message is changed_?